src/libYARP_OS/include/yarp/os/Type.h

```diff
@@ -9,84 +9,88 @@
 #ifndef YARP_OS_TYPE_H
 #define YARP_OS_TYPE_H
 
-#include <yarp/os/Searchable.h>
+#include <yarp/os/api.h>
+
 #include <string>
 
 namespace yarp {
-    namespace os {
-        class Type;
-        class Property;
-    }
-}
+namespace os {
+
+class Property;
+class Searchable;
+class Value;
 
-class YARP_OS_API yarp::os::Type {
+class YARP_OS_API Type
+{
 public:
+
+/** @{ */
+
+    /**
+     * @brief Constructor.
+     */
     Type();
 
-    Type(const Type& alt);
+    /**
+     * @brief Copy constructor.
+     *
+     * @param rhs the Type to copy
+     */
+    Type(const Type& rhs);
 
+    /**
+     * @brief Destructor.
+     */
     virtual ~Type();
 
-    const Type& operator =(const Type& alt);
-
-    static Type byName(const char *name) {
-        Type t;
-        t.name = name;
-        return t;
-    }
-
-    static Type byName(const char *name, const char *name_on_wire) {
-        Type t;
-        t.name = name;
-        t.name_on_wire = name_on_wire;
-        return t;
-    }
-
-    static Type byNameOnWire(const char *name_on_wire) {
-        Type t;
-        t.name = "yarp/bottle";
-        t.name_on_wire = name_on_wire;
-        return t;
-    }
-
-    static Type anon() {
-        return Type();
-    }
-
-    std::string getName() const {
-        return name;
-    }
-
-    std::string getNameOnWire() const {
-        return name_on_wire;
-    }
-
-    bool hasName() const {
-        return name!="";
-    }
-
-    bool isValid() const {
-        return hasName();
-    }
-
-    std::string toString() const {
-        if (name_on_wire!="") {
-            return name + ":" + name_on_wire;
-        }
-        if (name!="") return name;
-        return "null";
-    }
+    /**
+     * Copy assignment operator.
+     *
+     * @param rhs the Type to copy
+     * @return this object
+     */
+    Type& operator=(const Type& rhs);
+
+/** @} */
+/** @{ */
+
+    std::string getName() const;
```

Comment:
It would be good to add the documentation also for the other methods :wink:

Reply:
It would be good, if someone knew what they are supposed to do! :laughing: